packages/react-native/src/components/3DS/ThreeDSecureModal.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useContext } from "react";
+import { Modal, TouchableOpacity, Text, View, StyleSheet } from "react-native";
+import { ThreeDSecureModalProps, ThreeDSecureState } from "./types";
+import { ThreeDSecureFrame } from "./ThreeDSecureFrame";
+import { defaultModalStyles as defaultStyles } from "./styles";
+import { ThreeDSecureContext } from "./context";
+
+export const ThreeDSecureModal = ({config, style}: ThreeDSecureModalProps) => {
+
+  const state: ThreeDSecureState | null = useContext(ThreeDSecureContext);
+
+  if (!state) {
+    throw new Error("ThreeDSecure.Modal must be used within an Evervault ThreeDSecure provider component");
+  }
+
+  const close3DS = async () => {
+    if (state?.session) {
+      await state.session.cancel();
+    };
+  };
+
+  if (state.session && state.displayModal) {
```

Comment:
We should be able to remove the check for `state.displayModal` now that this will always be a child of the provider which has it's own check for that .

Reply:
Good shout, updated!